patches/tModLoader/Terraria/ModLoader/ProjectileLoader.cs

```diff
@@ -364,9 +364,11 @@ public static bool PreKill(Projectile projectile, int timeLeft)
 		return result;
 	}
 
+	[Obsolete]
 	private static HookList HookKill = AddHook<Action<Projectile, int>>(g => g.Kill);
 
-	public static void Kill(Projectile projectile, int timeLeft)
+	[Obsolete("Renamed to OnKill")]
+	public static void Kill_Obsolete(Projectile projectile, int timeLeft)
```

Comment:
We want to call `Kill_Obsolete` from `OnKill` here, keeps things nice and contained. Note that in some cases it may be necessary to interleave the mod and globals calls from the old and new hooks to preserve behaviour.

Reply:
K, that has been moved.